src/main/java/org/wso2/carbon/module/swiftiso20022/JsonToMT103Validator.java

```diff
@@ -0,0 +1,125 @@
+                this.log.error(ConnectorConstants.ERROR_INVALID_PAYLOAD, e);
+                ConnectorUtils.appendErrorToMessageContext(
+                        messageContext, ConnectorConstants.INVALID_REQUEST_PAYLOAD, e.getMessage());
+                super.handleException(ConnectorConstants.ERROR_INVALID_PAYLOAD, messageContext);
+            }
+
+            // validate values of the JSON object
+            this.log.debug("Validating payload parameters");
+            ValidationResult validationResponse = validateRequestPayload(payload.get());
+            if (validationResponse.isInvalid()) {
+                this.log.error(validationResponse.getErrorMessage());
+                ConnectorUtils.appendErrorToMessageContext(messageContext, validationResponse.getErrorCode(),
+                        validationResponse.getErrorMessage());
+                super.handleException(ConnectorConstants.ERROR_VALIDATION_FAILED, messageContext);
+            }
+        } else {
+            this.log.error(ConnectorConstants.ERROR_MISSING_PAYLOAD);
+            ConnectorUtils.appendErrorToMessageContext(messageContext, ConnectorConstants.MISSING_REQUEST_PAYLOAD,
+                    ConnectorConstants.ERROR_MISSING_PAYLOAD);
+            super.handleException(ConnectorConstants.ERROR_MISSING_PAYLOAD, messageContext);
+        }
+    }
+
+    private ValidationResult validateRequestPayload(String payloadString) {
+
+        // build JSONObject from the payload string
+        JSONObject requestPayload = new JSONObject(payloadString);
+
+        // validate block 01
+        ValidationResult blockValidationResult =
+                JsonToMt103Utils.validateBlock01(requestPayload.optJSONObject(MT103Constants.BLOCK01));
+        if (blockValidationResult.isInvalid()) {
+            return blockValidationResult;
+        }
+
+        // validate block 02
+        blockValidationResult = JsonToMt103Utils.validateBlock02(requestPayload.optJSONObject(MT103Constants.BLOCK02));
+        if (blockValidationResult.isInvalid()) {
+            return blockValidationResult;
+        }
+
+        // validate block 03
+        blockValidationResult = JsonToMt103Utils.validateBlock03(requestPayload.optJSONObject(MT103Constants.BLOCK03));
+        if (blockValidationResult.isInvalid()) {
+            return blockValidationResult;
+        }
+
+        // validate block 04
+        blockValidationResult = JsonToMt103Utils.validateBlock04(requestPayload.optJSONObject(MT103Constants.BLOCK04));
+        if (blockValidationResult.isInvalid()) {
+            return blockValidationResult;
+        }
+
+        // validate block 05
+        // validation result is returned since block 05 is the last block
+        return JsonToMt103Utils.validateBlock05(requestPayload.optJSONObject(MT103Constants.BLOCK05));
+    }
+
+    // validate JSON object with the defined request model
+    private void validatePayloadWithMT103JsonModel(String payload) throws JsonSyntaxException {
```

Comment:
Shall we add complete method comments

Reply:
Added in commit 96d4dc498094daee16277501378ee52fbb484dd1

Reply:
The JsonSyntaxException is thrown when Gson attempts to read a malformed JSON string. If it is our intention, shall we modify the method comment and explain the exact exception throwing reason?